lib/algolia/algoliaUtils.js

```diff
@@ -21,26 +31,66 @@ exports.cacheContentAlias = function cacheContentAlias(propPath) {
     return contentAlias;
 };
 
-exports.getRecordObject = function getRecordObject(filename, folderPath, basePath, contentAlias) {
+exports.getSectionContent = function getSectionContent(headers, pageContent, index, fileRecords) {
+    if (headers.length < 2) return pageContent;
+
+    let followingContent;
+
+    if (index === 0) {
+        followingContent = pageContent;
+    } else {
+        const lastSectionContent = fileRecords[index - 1].content;
+        followingContent = pageContent.split(lastSectionContent)[1];
+        if (index === headers.length - 1) return followingContent;
+    }
+    return followingContent.split(headers[index + 1])[0];
+};
+
+exports.getRecordObject = function getRecordObject(
+    filename,
+    folderPath,
+    basePath,
+    contentAlias,
+    getSectionContent,
+    splitSectionLevel = false
+) {
     const link = url
         .pathToFileURL(path.resolve(folderPath, filename))
         .toString()
         .split(basePath)[1]
         .slice(0, -4);
 
     const fileContent = fs.readFileSync(path.resolve(folderPath, filename)).toString();
-
     const platform = getPlatform(link);
-    const fileRecord = {
-        title: getTitle(fileContent),
-        link,
-        keywords: getKeywords(fileContent),
-        headings: getHeadings(fileContent),
-        content: getCleanedContent(fileContent, contentAlias),
-        platformName: platform,
-        objectID: link
-    };
-    return fileRecord;
+    const headers = getHeadings(fileContent);
+
+    if (splitSectionLevel) {

```

Comment:
```suggestion
  const result = splitSectionLevel ? getHeadings(fileContent) : fileContent
result.forEach(()=>{
const fileRecord = {...}
fileRecords.push(fileRecord)

})
return fileRecords
```

Reply:
check if the code duplication is avoidable

Reply:
Will refactor it